internal/revision/scanner.go

```diff
@@ -0,0 +1,124 @@
+package revision
+
+import (
+	"bufio"
+	"io"
+	"unicode"
+)
+
+var zeroRune = rune(0)
+
+// scanner represents a lexical scanner.
+type scanner struct {
+	r *bufio.Reader
```

Comment:
To spend more time working on that in need a bette argument than : 

> The only functionality your are using from a bufio.Reader is the readRune and unreadRune so using a full bufio.Reader cannot be justified.

If it's performance issue I'm ok with that, understandability as well but if it's only because I use this object for a subset of its functionalities what's wrong with that ? 

But I agree with the fact to rewrite that to have something better as with points you mentioned elsewhere